packages/grpc-js/src/server-call.ts

```diff
@@ -396,7 +396,8 @@ export class Http2ServerCallStream<
   ResponseType
 > extends EventEmitter {
   cancelled = false;
-  deadlineTimer: NodeJS.Timer = setTimeout(() => {}, 0);
+  deadlineTimer: NodeJS.Timer | null = null
```

Comment:
The whole point of having a noop timer instead of `null` here is to simplify the types, to avoid needing to check `if (this.deadlineTimer)` when acting on it. What exactly is the improvement gained by making this change?

Reply:
Point is to increase performance. by creating an empty timer we allocate underlaying resources that need to be further cleaned up. This is in the hot path and by removing the empty timer I was able to get ~5% perf increase on unary request throughput